design/Exceptions.md

```diff
@@ -0,0 +1,53 @@
+All error cases should be individually defined by the spec, on a case-by-case
+basis, as either:
+
+* synchronously throwing a JS exception, or
+* occurring asynchronously - causes a WebGPU object to become internally null,
+  an operation to no-op, or a `Promise` to throw. (Plus an error log entry.)
+
```

Comment:
It would be good if there was verbiage somewhere that even through the guidelines state that certain errors may or may not be synchronous depending on the situation, individual errors need to be either one or the other.